### PR TITLE
Multi tenancy/cluster role scope

### PR DESCRIPTION
**Pull request by chenqianfzh**

### CODE REVIEW

File: pkg/registry/rbac/clusterrole/policybased/storage.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (r *Storage) NamespaceScoped() bool {
 }
 
 func (r *Storage) TenantScoped() bool {
-	return false
+	return true
```
Comment (pdgetrf) [inline pkg/registry/rbac/clusterrole/policybased/storage.go]: what does this tenant scoped mean and why is it changed to true now?

Comment (pdgetrf) [reply pkg/registry/rbac/clusterrole/policybased/storage.go]: and if it's always going to be tenant-scoped, do we still need this boolean?

File: staging/src/k8s.io/client-go/informers/rbac/v1/clusterrole.go [hunk 2]
```diff
@@ -51,23 +52,31 @@ func NewClusterRoleInformer(client kubernetes.Interface, resyncPeriod time.Durat
...
 
+func NewClusterRoleInformerWithMultiTenancy(client kubernetes.Interface, resyncPeriod time.Duration, indexers cache.Indexers, tenant string) cache.SharedIndexInformer {
+	return NewFilteredClusterRoleInformerWithMultiTenancy(client, resyncPeriod, indexers, nil, tenant)
+}
+
 // NewFilteredClusterRoleInformer constructs a new informer for ClusterRole type.
 // Always prefer using an informer factory to get a shared informer instead of getting an independent
 // one. This reduces memory footprint and number of connections to the server.
 func NewFilteredClusterRoleInformer(client kubernetes.Interface, resyncPeriod time.Duration, indexers cache.Indexers, tweakListOptions internalinterfaces.TweakListOptionsFunc) cache.SharedIndexInformer {
+	return NewFilteredClusterRoleInformerWithMultiTenancy(client, resyncPeriod, indexers, tweakListOptions, "default")
```
Comment (pdgetrf) [inline staging/src/k8s.io/client-go/informers/rbac/v1/clusterrole.go]: metav1.TenantDefault?

and do we still want default tenant?

File: staging/src/k8s.io/client-go/listers/rbac/v1/clusterrole.go [hunk 3]
```diff
@@ -63,3 +67,52 @@ func (s *clusterRoleLister) Get(name string) (*v1.ClusterRole, error) {
 	}
 	return obj.(*v1.ClusterRole), nil
 }
+
+// ClusterRoles returns an object that can list and get ClusterRoles.
+func (s *clusterRoleLister) ClusterRoles() ClusterRoleTenantLister {
+	return clusterRoleTenantLister{indexer: s.indexer, tenant: "default"}
```
Comment (pdgetrf) [inline staging/src/k8s.io/client-go/listers/rbac/v1/clusterrole.go]: same question here about "default" tenant

File: pkg/registry/rbac/clusterrole/policybased/storage.go (hunk 1 above)
Comment (chenqianfzh) [reply pkg/registry/rbac/clusterrole/policybased/storage.go]: This value, together with the func NamespaceScoped(), indicate the scope of a resource type:
TenantScoped(): true, NamespaceScoped: true -> NamespaceScoped
TenantScoped(): true, NamespaceScoped: false -> TenantScoped
TenantScoped(): false, NamespaceScoped: false -> ClusterScoped

File: staging/src/k8s.io/client-go/listers/rbac/v1/clusterrole.go (hunk 3 above)
Comment (chenqianfzh) [reply staging/src/k8s.io/client-go/listers/rbac/v1/clusterrole.go]: That is the backward-compatible version, where the tenant value is set to "default".

There is new version, ClusterRolesWithMultiTenancy(), which allows the "tenant" to take any value.

File: staging/src/k8s.io/client-go/informers/rbac/v1/clusterrole.go (hunk 2 above)
Comment (chenqianfzh) [reply staging/src/k8s.io/client-go/informers/rbac/v1/clusterrole.go]: That is the backward-compatible version, where the tenant value is set to "default".

There is new version, ClusterRolesWithMultiTenancy(), which allows the "tenant" to take any value.

**Pull request merged**